test/check-machines-migrate

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2021 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import sys
+
+# import Cockpit's machinery for test VMs and its browser test API
+TEST_DIR = os.path.dirname(__file__)
+sys.path.append(os.path.join(TEST_DIR, "common"))
+sys.path.append(os.path.join(os.path.dirname(TEST_DIR), "bots/machine"))
+
+from machineslib import VirtualMachinesCase  # noqa
+from testlib import wait, test_main  # noqa
+from machinesxmls import POOL_XML  # noqa
+
+
+class TestMachinesMigration(VirtualMachinesCase):
+    provision = {
+        "machine1": {"address": "10.111.113.1/20", "dns": "10.111.112.100"},
+        "machine2": {"address": "10.111.113.2/20", "dns": "10.111.112.100"},
+    }
+
+    def prepare(self, machine1, machine2):
+        machine1.execute("echo '10.111.113.2 machine2' >> /etc/hosts")
+        machine1.execute("echo '10.111.113.1 machine1' >> /etc/hosts")
+        machine2.execute("echo '10.111.113.2 machine2' >> /etc/hosts")
+        machine2.execute("echo '10.111.113.1 machine1' >> /etc/hosts")
+
+        machine2.execute("sed -i 's/127.0.1.1.*/127.0.1.1 machine2 machine2/' /etc/hosts")
+        machine2.execute("hostnamectl set-hostname machine2")
+        machine2.execute("systemctl restart libvirtd")
+
+        # setup key authentication
+        machine1.execute("mkdir -m 0700 -p $HOME/.ssh")
+        machine1.execute("ssh-keygen -t rsa -N '\' -f $HOME/.ssh/id_rsa")
+        machine1.execute("echo 'StrictHostKeyChecking no' > $HOME/.ssh/config")
+        machine1PubKey = machine1.execute("cat $HOME/.ssh/id_rsa.pub")
+        machine2.execute("echo '{0}' >> /root/.ssh/authorized_keys".format(machine1PubKey))
+
+        # Setup the default storage pool on the destination host
+        # This is normally automatically created when creating the first VM but in this case our dest VM was never populated
```

Comment:
When you migrate a VM with a disk defined thru storage pool. Then the destination VM has to have the same storage pool (with the same name), otherwise libvirt would not know where to store the transferred disk. 

And in this case, the source host had the "default" storage pool, but destination doesn't have any storage pool, so we need to create one on destination too.

Maybe it's kinda unbreak my setup, but there is no way how to solve this problem in the UI, since cockpit on source host would not know anything about destination host, about storage environment there. So we just have to rely on the that that prepared the destination for the migration. 

This is one of the things which I want to discuss with @garrett about how to let user know that storage "environment" has to be similar on destination compared to source. (i.e. when you are migrating a VM with a disk from storage pool "myPool1", then on destination there also needs to be storage pool named "myPool1". Or if you are migrating a disk with path /home/skobyda/disks/mydisk.qcow2, then on the destination there needs to be a directory /home/skobyda/disks/ too). Perhaps we could make sense of error message it return in such cases, and give a hint: This usually happens when there is no storage pool named "myPool1" or something similar.

Reply:
I see -- so the *pool* must exist, but not the *volume* yet? Then let's please add a test that tries to migrate a VM with the remote pool not existing, and one where the remote pool *and* volume of the same name already exists. In both cases it should fail gracefully with some usable error message and not stop/destroy the local VM.

Reply:
Added a test case